lib/active_admin/resource_controller/decorators.rb

```diff
@@ -48,18 +48,47 @@ def collection_decorator
         end
       end
 
+      # This class decorates a collection of objects delegating
+      # mehods to behave like an Array. It's used to decouple ActiveAdmin
+      # from Draper and thus being able to use PORO decorators as well.
+      #
+      # It's implementation is heavily based on the Draper::CollectionDecorator
+      # https://github.com/drapergem/draper/blob/aaa06bd2f1e219838b241a5534e7ca513edd1fe2/lib/draper/collection_decorator.rb
+      class CollectionDecorator
+        # @return the collection being decorated.
+        attr_reader :object
+
+        # @return [Class] the decorator class used to decorate each item, as set by {#initialize}.
+        attr_reader :decorator_class
+
+        array_methods = Array.instance_methods - Object.instance_methods
+        delegate :==, :as_json, *array_methods, to: :decorated_collection
+
+        def initialize(object, with:)
+          @object = object
+          @decorator_class = with
+        end
+
+        def self.name
+          'ActiveAdmin::CollectionDecorator'
+        end
```

Comment:
I think it's not a good practice to override `self.name` for non anonymous classes. But I agree that the description looks better with this name. So maybe we can move the class to its own `lib/active_admin/collection_decorator.rb` and nest it directly under `ActiveAdmin` so that it has this better name.

Reply:
Gladly! Is it alright if I require the new file form `lib/active_admin/resource_controller.rb` or did you have anywhere else in mind?

Reply:
Sure!